docs/docs/learn/concepts/storage/notes.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Understanding Notes
+description: Core knowledge of Notes and how they work
+---
+
+Most prominant blockchain networks don't have privacy at the protocol level. Aztec contracts can define public and private functions, that can read/write public and private state.
+
+For private state we need encryption and techniques to hide information about state changes. For private functions, we need local execution and proof of correct execution.
+
+### Some context
+- public functions and storage work much like other blockchains in terms of having dedicated storage slots
+- Private functions are executed locally with proofs generated for sound execution, and commitments to private variable updates are stored using append-only trees.
+- "Note" types are part of aztec.nr, a framework that facilitates use of Aztec's different storage trees to achieve private variables. Note: there are more uses beyond this.
+
+This article will restrict its focus to how private variables are implemented with Notes and relevant storage trees. NB: these components can be used in a variety of ways to implement different attributes such as: singletons, unique value, or private sharing.
+
+#### Side-note about execution
+Under the hood, the Aztec protocol handles some important details around public and private function calls. Notably around calls between them and relevant delays to avoid race conditions around state reads/writes (more [here](../pxe/acir_simulator#simulating-functions)).
+Whilst it just works for us under the hood, a detailed explanation of the transaction lifecycle can be found [here](../transactions#simple-example-of-the-private-transaction-lifecycle).
+
+## Private state variables in Aztec
+State variables in an Aztec contract are defined inside a struct specifically named `Storage`, and must satisfy the [Note Interface](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_interface.nr) and contain a [Note header](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_header.nr).
+
+The Note header struct contains the contract address which the value is effectively siloed to, a nonce to ensure unique Note hashes, and a storage "slot" (or ID) to associate multiple notes.
+
+A couple of things to unpack here:
+
+#### Storage "slot"
+Storage slots are more literal for public storage, a place where a value is stored. For private storage, a storage slot is logical (more [here](./storage_slots#private-state-slots---slots-arent-real)).
+
+#### Silos
+The address of the contract is included in a Note's data to ensure contracts don't arrive at the same hash with an identical variable. This is handled in the protocol's execution.
+
+### Note types
+There is more than one Note type, but the `Set` type is most useful for private variables. There are also `Singleton` and `ImmutableSingleton` types.
```

Comment:
Avoid opinions. `Set` isn't more or less useful than others, they're used for different reasons
```suggestion
There is more than one Note type, such as `Set` `Singleton` and `ImmutableSingleton` types.
```

Reply:
Rephrased.